numba/np/arrayobj.py

```diff
@@ -981,7 +990,11 @@ def prepare(self):
         for i in self.indexers:
             i.prepare()
         # Compute the resulting shape
-        self.indexers_shape = sum([i.get_shape() for i in self.indexers], ())
+        shape_list = [i.get_shape() for i in self.indexers]
+        one = self.context.get_constant(types.intp, 1)
+        for i in self.newaxes:
+            shape_list.insert(i, (one,))
+        self.indexers_shape = sum(shape_list, ())
```

Comment:
Is this correct? I think this might trigger an issue present here:
```python
from numba import njit
import numpy as np

@njit
def foo(a):
    return a[np.newaxis, :, np.newaxis,  ..., np.newaxis]

def gen(n):
    return np.arange(np.prod(range(1, n + 1))).reshape((*range(n, 0, -1)))

x = gen(4)
got, expected = foo(x), foo.py_func(x)
np.testing.assert_allclose(got, expected)
```

Reply:
Resolved in [be5d0cc](https://github.com/numba/numba/pull/8513/commits/be5d0ccbbff8b33615fc6e2a6eb88c104fc5dbf2)